ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -2947,6 +2948,10 @@ private RelNode genTableLogicalPlan(String tableAlias, QB qb) throws SemanticExc
           if (AcidUtils.isNonNativeAcidTable(tabMetaData, false)) {
             virtualCols.addAll(tabMetaData.getStorageHandler().acidVirtualColumns());
           }
+          if (tabMetaData.isNonNative() && tabMetaData.getStorageHandler().areSnapshotsSupported() &&
+              isBlank(tabMetaData.getMetaTable())) {
```

Comment:
why is this check needed `isBlank(tabMetaData.getMetaTable())`?

Reply:
In case of metadata queries the `Table` object has a metaTable field which alters the meaning of SnapshotId
https://issues.apache.org/jira/browse/HIVE-25457 introduced the metaTable field.